### PR TITLE
MST Setup Completed & Business Related Models Created & mst-persist

### LINKED ISSUE #25: Mobx Integration
Integrate Mobx to yelp app and create models and necessary action for Businesses,Business Details, Events.

### LINKED ISSUE #27: Mobx Persist Store
Create Mobx Store that persists using [mst-persist](https://www.npmjs.com/package/mst-persist)

### PR DESCRIPTION
**Pull request by Naveen-gali**
### What does this PR do?

- Integrate Mobx to yelp Project
- Adds  the `Business` related models to yelp app.
- Sets up the Persist Store for dummy count store.

### Related Issues

- closes #25 
- closes #27 

### How to test

* Clone the Repository.
* Switch to `dev-mobxSetupAndModelsIntegration` branch.
* Do `yarn` to install the dependencies.
* Do `yarn run build-stage-ios` for `ios` or `yarn run build-stage-android` for `android`.
* A Counter along with 2 buttons will come up as shown in below attached screen recordings.
     * `increment` button to increase counter.
     * `decrement` button to decrease counter.
* Increase count by using increment button and try relaunching app by removing from **recent apps** as well.
* Count will start from where left.

#### Android Screen Recording
https://user-images.githubusercontent.com/122969303/228789095-1716c737-ce8b-4cdf-b6a1-da098084a56f.mp4

#### Ios Screen Recording.
https://user-images.githubusercontent.com/122969303/228788728-48ebcdbc-fdee-436e-b410-f763c843d838.mp4

### Checklist
- [x] I have tested this code in both IOS and Android.
- [x] I have attached the necessary ScreenShots or Screen Recordings.
- [x] I have tested this code locally and verified that it works as intended.
- [x] I have checked the code for formatting, syntax, and other best practices.
- [x] I have ensured that my changes do not break any existing functionality.

### Additional Notes

- [Mobx State Tree](https://mobx-state-tree.js.org/intro/welcome)
- [Async Storage](https://react-native-async-storage.github.io/async-storage/docs/install/)
- [Link to see the `protect` and `unprotect` usage](https://codesandbox.io/s/vibrant-frog-iud38?file=/src/App.tsx)

### Packages Added (if any)

| Package | Reason |
| ------- | ------ |
| mobx         | State Management        |
|mobx-react-lite | Lighter implementation of React Mobx |
|mobx-state-tree | Observable State Tree |
|mst-persist| Persisted State Management using Mobx |
|@react-native-async-storage/async-storage | Async Storage for storing Mobx Store |

### CODE REVIEW

File: src/models/RootStore.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const store = types
+  .model('RootStoreModel')
+  .props({
+    count: types.number,
+  })
+  .views(self => ({
+    get Count(): number {
+      return self.count;
+    },
+    set Count(val: number) {
```
Comment (prajnab-mm) [inline src/models/RootStore.ts]: If we are incrementing and decrementing using actions, then why do we need this?

File: App.tsx [hunk 2]
```diff
@@ -2,18 +2,32 @@ import React from 'react';
 import {NavigationContainer} from '@react-navigation/native';
 import {RootNavigator} from './src/navigation/RootNavigator';
 import RNBootsplash from 'react-native-bootsplash';
+import {store, RootStoreContext} from './src/models';
```
Comment (prajnab-mm) [inline App.tsx]: Sort the imports

Comment (prajnab-mm) [reply App.tsx]: Also, Can we move App.tsx within src folder?

Comment (Naveen-gali) [reply App.tsx]: ok i will fix it.

File: App.tsx [hunk 3]
```diff
@@ -2,18 +2,32 @@ import React from 'react';
...
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import {persist} from 'mst-persist';
+
+const RootStore = store.create({
```
Comment (prajnab-mm) [inline App.tsx]: Move these logic within Model folder. Either in index.ts or RootStore.ts

File: App.tsx [hunk 4]
```diff
@@ -2,18 +2,32 @@ import React from 'react';
...
+  count: 0,
+});
 
 function App(): JSX.Element {
+  persist('RootStore', RootStore, {
```
Comment (prajnab-mm) [inline App.tsx]: We should just call a method called setupStore from within model and it should take care of all this

Comment (Naveen-gali) [reply App.tsx]: Prajna Can you share me something to read about this setupStore, cause i can't find it and it is mentioned in the documentation and the example as well to follow this pattern.

https://github.com/agilgur5/react-native-manga-reader-app/blob/master/App.js#L11

**Review by prajnab-mm — APPROVED** [review]

File: App.tsx (hunk 4 above)
Comment (prajnab-mm) [reply App.tsx]: As discussed, the code is fine. Just wanted you to move it to model folder as its a code related to model. If we start adding initialisation codes of everything in App.tsx then the file length will be huge and hard to read

**Pull request merged**